rest_framework/renderers.py

```diff
@@ -449,6 +449,11 @@ def render(self, data, accepted_media_type=None, renderer_context=None):
         else:
             template_name = style['template_pack'].strip('/') + '/' + style['base_template']
 
+        # Remove hidden fields from serializer before rendering
+        for field in form.fields:
+            if isinstance(form.fields[field], serializers.HiddenField):
+                form.fields.pop(field)
```

Comment:
I don't think that mutating the serializer instance is the right way to approach this - unexpected side effects.